contracts/contracts/oracle/OracleRouter.sol

```diff
@@ -34,14 +38,35 @@ abstract contract OracleRouterBase is IOracle {
         require(_feed != FIXED_PRICE, "Fixed price feeds not supported");
         (, int256 _iprice, , , ) = AggregatorV3Interface(_feed)
             .latestRoundData();
-        uint256 _price = uint256(_iprice);
+        uint8 decimals = getDecimals(asset);
+
+        uint256 _price = uint256(_iprice).scaleBy(18, decimals);
         if (isStablecoin(asset)) {
             require(_price <= MAX_DRIFT, "Oracle: Price exceeds max");
             require(_price >= MIN_DRIFT, "Oracle: Price under min");
         }
         return uint256(_price);
     }
 
+    function getDecimals(address _asset)
+        internal
+        view
+        returns (uint8)
+    {
+        require(decimalsCache[_asset] > 0, "Oracle: Decimals not cached");
```

Comment:
This does two reads. Let's use a temp var

Reply:
yes, thanks... done